File: apps/backend/hash_passwords.js
// MongoDB script to hash all user passwords as "password"
// Run this in mongosh or MongoDB Compass

// Hash function (using bcrypt-like hash)
function hashPassword(password) {
  // This is a simple hash for demo purposes - in production use proper bcrypt
  return password; // For demo, we're just setting plain text - DON'T DO THIS IN PRODUCTION
}

// Connect to your database
use AgriVote;

// Update all users to have hashed "password"
db.users.updateMany(
  {}, // Update all documents
  [
    {
      $set: {
        hashed_password: hashPassword("password")
      }
    }
  ]
);

// Alternatively, if you want to use a proper bcrypt hash, run this Node.js script:

// const bcrypt = require('bcrypt');
// const { MongoClient } = require('mongodb');
//
// async function hashPasswords() {
//   const client = new MongoClient(process.env.MONGODB_URI || 'mongodb://localhost:27017');
//   await client.connect();
//
//   const db = client.db('AgriVote');
//   const users = db.collection('users');
//
//   const userDocs = await users.find({}).toArray();
//
//   for (const user of userDocs) {
//     const hashedPassword = await bcrypt.hash('password', 10);
//     await users.updateOne(
//       { _id: user._id },
//       { $set: { hashed_password: hashedPassword } }
//     );
//     console.log(`Updated password for user: ${user.email}`);
//   }
//
//   await client.close();
// }
//
// hashPasswords();

File: apps/backend/hash_passwords.py
#!/usr/bin/env python3
"""
Python script to hash all user passwords in MongoDB using the same algorithm as auth_routes.py
This script updates all existing users to have properly bcrypt-hashed "password"
"""

import os
import sys
import bcrypt
from pymongo import MongoClient
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

def get_database():
    """Connect to MongoDB and return the AgriVote database"""
    mongodb_url = os.getenv("MONGODB_URL", "mongodb://localhost:27017")
    db_name = os.getenv("DATABASE_NAME", "AgriVote")
    client = MongoClient(mongodb_url)
    return client[db_name]

def main():
    """Update all user passwords with bcrypt hash"""
    try:
        # Use direct bcrypt hashing (compatible with FastAPI bcrypt)
        password_to_hash = b"password"
        salt = bcrypt.gensalt()
        hashed_password = bcrypt.hashpw(password_to_hash, salt).decode('utf-8')

        print(f"Hashing password '{password_to_hash}'...")

        # Connect to database
        db = get_database()
        users_collection = db.users

        # Update all users with the hashed password
        result = users_collection.update_many(
            {},  # Update all documents
            {"$set": {"hashed_password": hashed_password}}
        )

        print(f"✅ Successfully updated {result.modified_count} users")
        print(f"Bcrypt hash: {hashed_password}")

        # Verify one user
        sample_user = users_collection.find_one({}, {"email": 1, "hashed_password": 1})
        if sample_user:
            print("\nSample user updated:")
            print(f"  Email: {sample_user.get('email', 'N/A')}")
            print(f"  Hash: {sample_user.get('hashed_password', 'N/A')}")

        print("\n🎉 All passwords updated successfully!")
        print("Users can now login with password: 'password'")

    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)
    finally:
        if 'client' in locals():
            client.close()

if __name__ == "__main__":
    main()
